Remove schema data

benjeffery opened the pull request:
Stacked on #339 - clean diff at https://github.com/sgkit-dev/bio2zarr/pull/343/commits/9f727a1e604505a5759fd677b76ac423985676fe

benjeffery commented:
Rebased

jeromekelleher commented on CHANGELOG.md:
```diff
@@ -4,6 +4,8 @@
 
 - Add support for unindexed (and uncompressed) VCFs (#337)
 
+- Remove explicit sample, contig and filter lists from the schema (#343)
```
This is a breaking schema format change, so will require existing ICFs to be recreated. I listed these as "Breaking changes" below

jeromekelleher reviewed (commented):
Looks good, I think we just need to formalise the interface we're implementing a bit here.

benjeffery commented:
I've added a base class for the sources and removed the mapping base class from the icf source. I've kept contigs optional as plink doesn't have them in this PR.
